### ISSUE TITLE
Failed to determine boots: No data available v239.303-1

### ISSUE THREAD

**Issue by CarbonChauvinist** [systemd/systemd#11089]
**systemd version the issue has been seen with**
239.303-1

**Used distribution**
Linux 4.19.4-arch1-1-ARCH x86_64

**Expected behaviour you didn't see**
Expect to see list of boots when running journalctl --list-boots

**Unexpected behaviour you saw**
Failed to determine boots: No data available

**Steps to reproduce the problem**
Run journalctl --list-boots

**Comment by CarbonChauvinist** [comment]
journalctl --verify
https://ptpb.pw/XMBX

**Comment by CarbonChauvinist** [comment]
SYSTEMD_LOG_LEVEL=debug journalctl --verify
https://ptpb.pw/YzWC

SYSTEMD_LOG_LEVEL=debug journalctl --list-boots
https://ptpb.pw/OIwe

**Comment by CarbonChauvinist** [comment]
from /var/log/journal/19cf6c87e9104d3abe55b344266a78c3/ ran following command:
`find . -iname "*.journal" -exec journalctl --verify --file {} \;`
https://ptpb.pw/Wb85

from same location ran following command:
`find . -iname "*.journal\~" -exec journalctl --verify --file {} \;`
https://ptpb.pw/JLU-

Besides some unused data messages - this seems to be the only ?real failure? I can see

> 4c97a8: Invalid object                                                                                                                                                                   
File corruption detected at ./system@00057b7cfdff1979-c2ef630cf958fefd.journal~:4c97a8 (of 8388608 bytes, 59%).                                                                          
FAIL: ./system@00057b7cfdff1979-c2ef630cf958fefd.journal~ (Bad message)

**Comment by poettering** [comment]
any chance you can provide me with the full set of journal files that cause this?

**Comment by CarbonChauvinist** [comment]
AFAICT I've attached the one journal file that appears to cause the reported issue. If I remove this single file I'm able to --list-boots as expected. However, if I add the file back, I get the same reported error.

Here I copied the single file to another directory to test.

> SYSTEMD_LOG_LEVEL=debug journalctl --directory=./ --list-boots
Considering root directory './'.
Root directory ./ added.
Journal effective settings seal=no compress=no compress_threshold_bytes=8B
File .//system@0005753bb65373c3-a2209ab86b1168b0.journal~ added.
Whoopsie! We found a boot ID but can't read its last entry.
Failed to determine boots: No data available
Root directory ./ removed.
mmap cache statistics: 490 hit, 3 miss

But it seems to pass a verify?

> sudo SYSTEMD_LOG_LEVEL=debug journalctl --verify --file system@0005753bb65373c3-a2209ab86b1168b0.
journal~
Journal effective settings seal=no compress=no compress_threshold_bytes=8B
File system@0005753bb65373c3-a2209ab86b1168b0.journal~ added.
1041898: Unused data (entry_offset==0)
1041958: Unused data (entry_offset==0)
10419c8: Unused data (entry_offset==0)
PASS: system@0005753bb65373c3-a2209ab86b1168b0.journal~
mmap cache statistics: 12882932 hit, 7 miss

[data.zip](https://github.com/systemd/systemd/files/2678464/data.zip)
